### ISSUE TITLE
🖐️ Hide WithExec stdout in Dagger logs

### ISSUE THREAD

**Issue by ojroques** [dagger/dagger#8798]
### What happened? What did you expect to happen?

Hello,

I'm running Dagger in GitLab CI. I'm using it to request certificate to Hashicorp Vault. However Dagger prints the output of the request in plaintext in our GitLab CI, which includes the certificate key. Is there an option to specifically disable the output of a command in Dagger UI, without silencing everything else (with `-s`)?

Thanks.

**Comment by Xiot** [comment]
I just ran into this as well.
I did noticed that there is a `redirectStdout` option that redirects the output to a file, but the results still show up in the logs

**Comment by Xiot** [comment]
I just ended up using
```typescript
const result = await container
  .withExec(['sh', '-c', 'my-command.sh > /tmp/output.log')
  .file('/tmp/output.log')
  .contents()
```

Not the nicest, but it does act as a work around for me for now

**Comment by marcosnils** [comment]
adding some context here that @vito shared in our Discord server (https://discord.com/channels/707636530424053791/1301661744984494081/1303173319574687860) that the  implementation to make this happen seems to be trivial. I guess the most difficult part is how the API change looks like and how we could implement this in a way that's consistent.  @Xiot @ojroques happy to kick off the API discussion so we can take it to a good enough point where this can be implemented.

**Comment by Xiot** [comment]
Wouldn't the existing api with `redirectStdout` cover it?

```typescript
.withExec(['cmd', 'arg1', 'arg2'], {redirectStdout: '/tmp/output.log'})
```

Or do you mean more generically with something like
```typescript
.withExec(['cmd', 'arg1', 'arg2'], {suppressStdout: true})
```

**Comment by ojroques** [comment]
The first command with `redirectStdout` still prints the output to dagger TUI unfortunately, so I'd hope for a an option like in your second example.

**Comment by Xiot** [comment]
Correct. `redirectStdout` should cause the result from the command to be suppressed from the logs.  The point that I was referring to was that if that was 'fixed', then we wouldn't need an api change

**Comment by marcosnils** [comment]
> Correct. `redirectStdout` should cause the result from the command to be suppressed from the logs. The point that I was referring to was that if that was 'fixed', then we wouldn't need an api change

 this might seem reasonable. I'm pinging other people in this issue like @jedevc  @sipsma  to see how they feel about this idea.

**Comment by sipsma** [comment]
> > Correct. `redirectStdout` should cause the result from the command to be suppressed from the logs. The point that I was referring to was that if that was 'fixed', then we wouldn't need an api change
> 
> this might seem reasonable. I'm pinging other people in this issue like @jedevc @sipsma to see how they feel about this idea.

I'm not necessarily opposed. It would be a breaking change technically but I can see how it makes more sense in terms of behavior.

My broader concern is that all of these approaches would still result in the secret existing in the local cache (and remote cache if it's exported). If a secret like this is sensitive enough to be worth scrubbing from the TUI output it feels like you probably wouldn't want it stored in plaintext in the engine's cache either.

@ojroques @Xiot do your use cases require running a command that writes the private key to a file from a command in a `withExec`? Guessing so but any details on what commands you're relying on specifically could be helpful.

The above isn't a blocker on the potential change to `redirectStdout/err` behavior, but seems worth figuring out here.

**Comment by marcosnils** [comment]
> My broader concern is that all of these approaches would still result in the secret existing in the local cache (and remote cache if it's exported). If a secret like this is sensitive enough to be worth scrubbing from the TUI output it feels like you probably wouldn't want it stored in plaintext in the engine's cache either.

 According to what I've read in Discord there's currently two use-cases for this feature: 

1. @ojroques is looking for a way to scrub sensitive data from stdout
2. @Xiot  on the other hand just wants to remove a very long output from stdout as it makes reading the output quite cumbersome. 

I think that not printing the output when `redirectStdX` is set makes sense for option 2. For option 1, seems like we might need something different as it involves avoiding storing things in the cache as your're mentioning @sipsma ?

**Comment by Xiot** [comment]
Correct, I just want to hide the output since it just makes the logs harder to read. 

If you didn't want to make a breaking change, you could add a `suppressLogOutput` flag.

**Comment by benkeil** [comment]
any news?

**Comment by anton-slashm** [comment]
Having issues with this, too.
